### PR TITLE
Add support for phpSpec ^5.0

### PR DESCRIPTION
**Pull request by pamil**
I'm back with yet another major phpSpec version support :) Should be ready to tag (release date in the changelog is tomorrow 🎉).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,32 +9,22 @@ Installation
 
 Once you have installed PhpSpec (following the documentation on [the official website](http://www.phpspec.net)), add the extension requirement to your `composer.json`:
 
-Using phpspec 2.x,
```
Comment (pamil) [inline README.md]: Removed usage instructions for EOLed versions (2 and 3).

File: src/Akeneo/Runner/Maintainer/SkipExampleMaintainer.php [hunk 2]
```diff
@@ -66,7 +66,7 @@ public function getPriority(): int
      *
      * @return array
      */
-    protected function getRequirements(string $docComment): array
+    private function getRequirements(string $docComment): array
```
Comment (pamil) [inline src/Akeneo/Runner/Maintainer/SkipExampleMaintainer.php]: The class is final.

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 language: php
 
-php: [7.0, 7.1]
+php: [7.1, 7.2]
```
Comment (pamil) [inline .travis.yml]: phpSpec supports ^7.1, so should this extension

**Comment by pamil** [comment]
Ping 🏓

File: .travis.yml (hunk 3 above)
Comment (jmleroux) [reply .travis.yml]: You can remove 7.1 as we use 7.2 since 2.0

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,10 +1,14 @@
+# 4.0.x (2017-07-16)
```
Comment (jmleroux) [inline CHANGELOG.md]: Please update the date

**Comment by jmleroux** [comment]
Hello @pamil ,

Thnaks for the PR and sorry for the delay. I'll see with the team if we can approve it quickly and will keep you updated.

JM

File: composer.json [hunk 5]
```diff
@@ -30,7 +30,8 @@
     "minimum-stability": "stable",
     "extra": {
         "branch-alias": {
-            "dev-master": "3.0.x-dev",
+            "dev-master": "4.0.x-dev",
+            "dev-3.0": "3.0.x-dev",
```
Comment (damien-carcel) [inline composer.json]: Reminder => we need to create the `3.0` branch when merging this PR.

**Comment by jmleroux** [comment]
@pamil , the PR is approved. I will merge / tag as soon as you have fixed the last minor comments.

File: .travis.yml (hunk 3 above)
Comment (damien-carcel) [reply .travis.yml]: I agree with @pamil. We could very well update the dev-dependencies of Akeneo 2.x too, not just 2.3. I think we should let 7.1.

However, we can add 7.3 for the future, as it is now available on Travis (RC3 currently).

**Comment by pamil** [comment]
Great, thanks a lot! :)

**Comment by damien-carcel** [comment]
Thanks to you for your contribution, and again sorry we were so long to review it.

**Pull request merged**